Better testing for transforms + code which conforms to the style guide

willtebbutt opened the pull request:
Partially addresses #542. Doesn't currently do vector- or matrix-variate distributions for technical reasons.

Also sorts out indentation in `src/utilities/transform.jl`.

willtebbutt commented:
Current failures appear to be the usual statistical ones. I'll try to sort out tests for vector- and matrix-variate distributions this afternoon, then we should be good to go.

mohamed82008 commented:
Could a missing `Symmetric` wrapper be causing https://travis-ci.org/TuringLang/Turing.jl/jobs/428954293#L764? I cannot reproduce it on my machine so it could be floating point error causing some asymmetry in a matrix on which `cholesky` is called. If the matrix is not perfectly symmetric, it will trigger a positive definite error, unless wrapped by `Symmetric`.

willtebbutt commented:
I've added a number of tests using the `@test_broken` macro where I couldn't figure out the best way to check for consistency between the link function and `logpdf_with_trans` implementation using AD. These won't prevent CI from passing, but they will flag up in the test summary. This way, it will remind us that we have things that we need to fix, but doesn't halt development, and allows us to merge the code that I've already added in this PR (which I think is strictly better than what we had before).

willtebbutt commented:
Turns out that `LogMvNormal` wasn't being properly transformed, and our tests didn't cover this. I've now added an appropriate transformation for `LogMvNormal`, and added tests to ensure that if we take arbitrary points in the unconstrained domain and map them to the constrained domain using the appropriate inverse link function the `logpdf` is finite. These tests are statistical in nature (obviously you can't cover the entire space) but do a much better job than we were doing previously (which was nothing at all).

yebai commented:
Thanks, @willtebbutt. The code looks good to me!
